.github/workflows/tests.yml

```diff
@@ -34,13 +34,31 @@ jobs:
           persist-credentials: false
       - name: Set up Python
         uses: actions/setup-python@v6
+        id: setup_python
```

Comment:
I think we should have at least one job that runs with the latest dependencies (besides the byte-compiled job, which currently has its own problems.)

But we don't necessarily want that noise on PRs.

Let's use this job, which runs on pushes to main.

Could you revert the changes, and retitle this job to `windows-latest-dependencies` to make it self-documenting why it's not following the pattern?